### DISCUSSION TITLE
Why `groups ~ @request.auth.group` and not ` @request.auth.group ~ groups`

### DISCUSSION THREAD

**Discussion by xrd** [pocketbase/pocketbase#1697] [Q&A] [answered]
I've added a field to my user table called "group". And then my meetings table has "groups"

Group (for a user) might have a value like this: `friends`

Groups (for a meeting) might look like this: `friends, enemies`

In my API rules for meetings, I am using this filter for the list/search action. `groups ~ @request.auth.group`  And, that indeed seems to filter the meetings by the group that user is in.

But, shouldn't this be the opposite: ` @request.auth.group ~ groups`? To me, the latter reads like `PERMIT IF a user's group ("friends") is IN a meetings groups ("friends, enemies")`.

Is this me confusing the ordering of a LIKE clause where in SQL it is `COLUMN like PATTERN` versus a programming language where it is ` if ( "friends" in "friends, enemies") `?

I'm just making sure I am not missing something significant about this query and it is doing what I expect it to do.

**Comment by ganigeorgiev** [comment, marked as answer]
`~` and `!~` operators internally resolves to `LIKE` and `NOT LIKE`, so the correct order as you noted in your case should be `groups ~ @request.auth.group` (could be read also as "groups contains auth.group").

v0.11 introduced match-all (default) and any/"at least one of" operators (when prefixed with `?`) for all multi-value fields, so the above filter could be also written as:

```js
// at least one of the groups' id is equal to the auth.group
groups.id ?= @request.auth.group
```
